pkg/util/util.go

```diff
@@ -0,0 +1,16 @@
+package util
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const OVN_INTERCONNECT_CONFIGMAP_NAME = "ovn-interconnect-configuration"
+const OVN_NAMESPACE = "openshift-ovn-kubernetes"
+
+func GetInterConnectConfigMap(kubeClient kubernetes.Interface) (*corev1.ConfigMap, error) {
```

Comment:
@ricky-rav @JacobTanenbaum  for Hypershift, we must have a configmap PER hosted cluster.
What that means is that we need to use the namespace named after the hosted cluster, like: `clusters-${CLUSTER_NAME}`
We can use a logic similar to  https://github.com/openshift/cluster-network-operator/pull/1860/commits/856b92d743d4059194272161628d5e6bd99eae70#r1256628399 to determine that it is hypershift and something else to extrace the hosted cluster name.
cc @kyrtapz

Reply:
true true true! I start wondering about that an hour ago, I'm glad you are confirming it :) 
Thanks, Flavio!

Reply:
@flavio-fernandes @ricky-rav @JacobTanenbaum Looking at how the configmap gets created:
```
	if err := client.ClientFor("").CRClient().Create(context.TODO(), configMap); err != nil {
			return progressing, fmt.Errorf("could not create interconnect configmap: %w", err)
		}
```
we can see that it uses the default client (`ClientFor("")`). 
This means that with the current changes, in HyperShift, it will be created in the hosted cluster, not in the management.
If we keep it this way it means that we are already creating a confimap per hosted cluster in HyperShift.
I don't see a reason why we would need to move that to the management cluster but I haven't put much thought into it.

Reply:
That's even better news, I'll push the config map in the hosted cluster openshift-ovn-kubernetes namespace, then. Thanks, Patryk!

Reply:
mark as resolved